### ISSUE TITLE
Zoom leave and visibility of icon in symbol layer

### ISSUE THREAD

**Issue by weix-uber** [mapbox/mapbox-gl-js#7705]
**mapbox-gl-js version**:

### Question
I am using the 'symbol' layer + 'symbol-placement' = 'line' to display  arrows on a line. The example can be found in https://jsfiddle.net/waynexiao/mfzv8dsL/11/

The arrows show at zoom level 12, but not zoom level 11 and less. The zoom level is set at the line 24. At zoom 11, the length of the line is more 130px and the size of the arrow is about 12px x 12px.  There seems to be enough space to show arrows.

Is it possible to make the arrow to show at zoom level <= 11?

Thanks,

Wei

### Links to related documentation

https://jsfiddle.net/waynexiao/mfzv8dsL/11/

**Comment by mourner** [comment]
This looks like it's an issue with how the line anchors code takes icon size into account — it thinks the icons won't fit into the line even though `icon-size` is `0.045`. For the same reason, even on higher zooms, the gaps are pretty big even with `symbol-spacing` equal to `1` (pixel). Labeling this as a bug. @ChrisLoer any quick clues on this one?

**Comment by ChrisLoer** [comment]
Yeah, looks like the problem is that when we calculate the label length for anchor-spacing purposes, we take the max of the "icon" and "text" length, and then multiply them by the same "box scale" factor:

https://github.com/mapbox/mapbox-gl-js/blob/b43478c5d5e36376a9644ef195a7cd659332dd96/src/symbol/get_anchors.js#L29-L43

... but that scaling factor is actually just the scaling factor for the text, so the `icon-size` is effectively ignored:

https://github.com/mapbox/mapbox-gl-js/blob/b43478c5d5e36376a9644ef195a7cd659332dd96/src/symbol/symbol_layout.js#L216-L226

I think the fix is just to pass the icon box scale into `getAnchors` and use it to calculate the width of the icon.

cc @ansis

**Comment by weix-uber** [comment]
> Yeah, looks like the problem is that when we calculate the label length for anchor-spacing purposes, we take the max of the "icon" and "text" length, and then multiply them by the same "box scale" factor:
> 
> cc @ansis

Is the valid workaround to reduce the actual size of the icon and set 'icon-size' to 1?

Thanks!

**Comment by ChrisLoer** [comment]
> Is the valid workaround to reduce the actual size of the icon and set 'icon-size' to 1?

Yup, that's a good workaround!

**Comment by cfecherolle** [comment]
Hi. I seem to have the same issue as the OP but do not understand what the workaround would be.
I'm using the attached icon, which I tried in both SVG and PNG version and has a size of 500x500px.

**SVG version:**
![edge-right-arrow-icon](https://github.com/mapbox/mapbox-gl-js/assets/3169480/293b0f7d-8d42-4b35-9f98-58631c425aa3)

**PNG version:**
![edge-right-arrow-icon](https://github.com/mapbox/mapbox-gl-js/assets/3169480/b8d2f125-7708-4d6f-a72b-ff06437d169c)

**Layout code**
````
{
   "symbol-placement": "line-center",
   "icon-allow-overlap": true,
    "icon-image": edgeRightArrowIconId,
    "icon-size": 1,
}
```
In both cases, the icon is displayed in a very small size, much too small for my use case:
<img width="297" alt="image" src="https://github.com/mapbox/mapbox-gl-js/assets/3169480/1e074a59-cf00-422c-8625-e3f0162b0b47">

I tried using the `icon-size` property but as it heavily pixelates the icon, it does not give an acceptable result.
What can I do to have a bigger icon at the center of the line, but non-pixelated? Is this a bug?
